slamon_afm/routes/agent_routes.py

```diff
@@ -192,25 +196,19 @@ def post_tasks():
         current_app.logger.error("Invalid protocol in task response: {0}".format(protocol))
         abort(400)
 
-    result = ""
     try:
         task = db.session.query(Task).filter(Task.uuid == str(task_id)).one()
 
-        if task.claimed is None or (task.completed is not None or task.failed is not None):
-            current_app.logger.error("Incomplete task posted!")
-            abort(400)
-
-        if 'task_data' in data:
-            task.result_data = data['task_data']
-            task.completed = datetime.utcnow()
-            result = task.result_data
-        elif 'task_error' in data:
-            task.error = data['task_error']
-            task.failed = datetime.utcnow()
-            result = task.error
+        task.complete(
+            result_data=data['task_data'] if 'task_data' in data else None,
```

Comment:
I'm learning new things :see_no_evil: 
I'll definitely update it to make it more readable